src/main/webapp/script.js

```diff
@@ -38,12 +38,123 @@ function isSignedIn() {
   });
 }
 
-// Show the sign in button.
-function displaySignInButton() {
-  document.getElementById('sign-in-button-form').style.display = 'block';
+// Create the necessary elements and display the sign in page.
+function displaySignInPage() {
+  // Set the width of the content container.
+  setContentWidth('650px');
+
+  // Add background image to site. (This id is defined in CSS.)
+  document.body.id = 'body-background-image';
+
+  // Get the homepage "registration" block to and add elements.
+  const indexRegistrationBlock = document.getElementById('index-registration-block');
+  indexRegistrationBlock.style.display = 'block';
+
+  // Create h1 element as title for TravIS.
+  const titleElement = document.createElement('h1');
+  titleElement.innerText = 'TravIS: Travel Information System';
+  titleElement.className = 'index-registration-block-child';
+
+  // Create p element to describe TravIS.
+  const textElement = document.createElement('p');
+  textElement.innerText = 
+    'TravIS, your personal Travel Information System, allows you to plan ' +
+    'the optimal trip to wherever you want to visit.';
+  textElement.className = 'index-registration-block-child';
+  textElement.id = 'index-registration-block-p';
```

Comment:
Not required for the prototype this week. But I'd prefer to have these tags defined in the html page then shown/hidden by the javascript. Ideally there's division of responsibility between the html and javascript files. Where the html file should describe all the _contents_ of the page and the js file describes only the _behaviors_ of the page. Creating and adding all these tags here makes it harder to interpret the behavior (ie: what triggers a change what that change does) and harder to interpret what shows up on the page since this format is less readable (imo) than straight html.

I get that in this particular case it's a bit awkward because you don't know what to show while you're waiting for a response from the authserver. You can default to showing a blank or loading page and then show/hide the appropriate content based on how the authserver responds.

Reply:
Just saw your comment in the header.

I wouldn't worry too much about users seeing signed-in html content before they actually sign in, since that signed-in content won't reveal any users private information b/c you'd have to retrieve personal contents from the server first anyways. Plus this approach doesn't necessarily fix the problem b/c a user could view the script.js in the inspect tab as well as the html.

I'm not totally sure what the best strategy is for creating a foolproof sign-in barrier. This might be something to bring up with the VSE's. Maybe you could have a separate sign-in page and just re-direct to there if a user isn't signed in? or maybe there's other ways to conditionally load page content that doesn't require building it all up in js? Depending on the complexity of the problem this might be worth adding to the design doc too.

Reply:
Talked with Tristan on call -- decided to revert this back to HTML and use the JS just to change the `display` settings of different divs, depending on the user's sign-in status.

Reply:
One option may be to have the content be fetched from a servlet for the index page; this would prevent the user from seeing any content they were not supposed to see. However, for the prototype, changing the `display` status in the JS file should be sufficient.